modules/@apostrophecms/doc/index.js

```diff
@@ -347,6 +347,15 @@ module.exports = {
           submitted: 1,
           aposLocale: 1
         });
+        await self.db.createIndex({
+          type: 1,
```

Comment:
Without this index, the query optimizer guesses wrong for piece queries on specific aposDocIds, especially when they are numerous (it uses the type - plus - aposLocale index instead).